### PR TITLE
remove in-place swap on host for sharded in FD

### PR DESCRIPTION
**Pull request by ntarafdar**
This is what we talked about @arakhmati @cfjchu 

Now this test works : `pytest models/demos/falcon40b/tests/test_falcon_mlp.py::test_FalconMLP_inference[BFLOAT16-SHARDED-0.9987-tiiuae/falcon-40b-instruct-prefill-1-64-8chips]`

without having to make a separate host copy in python. 

CC: @TT-BrianLiu 

Also tacked on commit to do host side swap of entire width of a shard at a time. (instead of page by page)

### CODE REVIEW

File: tt_metal/impl/dispatch/command_queue.cpp [hunk 1]
```diff
@@ -857,14 +857,14 @@ void HWCommandQueue::enqueue_command(T& command, bool blocking) {
...
+void * convert_interleaved_to_sharded_on_host(const void* host, const Buffer& buffer, bool read = false) {
     const uint32_t num_pages = buffer.num_pages();
     const uint32_t page_size = buffer.page_size();
 
     const uint32_t size_in_bytes = num_pages * page_size;
 
-    void* temp = malloc(size_in_bytes);
-    memcpy(temp, host, size_in_bytes);
+    void* swapped = malloc(size_in_bytes);
+    memcpy(swapped, host, size_in_bytes);
```
Comment (tt-aho) [inline tt_metal/impl/dispatch/command_queue.cpp:867]: You don't need this memcpy here

Comment (tarafdarTT) [reply tt_metal/impl/dispatch/command_queue.cpp:867]: ah you're right, still an artifact of before :p

**Comment by tt-aho** [comment]
Something to consider is that this implementation currently duplicates the memory usage of the entire tensor when writing to device, which could be expensive for large tensors.

An optimization that would reduce memory usage but shouldn't affect perf is to potentially only remap the given chunk we are writing at a time, this way we can still do a contiguous write for the chunk while reducing total memory usage at a given time.

**Review by arakhmati — APPROVED** [review]

**Review by pgkeller — APPROVED** [review]
I'm not familiar w/ this code, don't see any issues.  Please have Austin review.

**Pull request merged**